components/eam/src/control/sat_hist.F90

```diff
@@ -82,6 +84,14 @@ module sat_hist
 
   real(r8), parameter :: rad2deg = 180._r8/pi            ! degrees per radian
 
+  logical :: flds_scanned = .false.
+  logical :: has_phys_srf_flds = .false.
+  logical :: has_phys_lev_flds = .false.
+  logical :: has_phys_ilev_flds = .false.
+  logical :: has_dyn_srf_flds = .false.
+  logical :: has_dyn_lev_flds = .false.
+  logical :: has_dyn_ilev_flds = .false.
```

Comment:
You might want to consider restricting the scope of these vars ("save" for flds_scanned, and have the other variables returned from the scan_flds function)